### PR TITLE
uffd: use userns_call() to execute ioctl(UFFDIO_API)

### PR DESCRIPTION
**Pull request by rppt**
In the recent kernels the userfaultfd support for FORK events is limited to
CAP_SYS_PTRACE. That causes the followong error when the ioctl(UFFDIO_API)
is executed from non-privilieged userns:

Error (criu/uffd.c:273): uffd: Failed to get uffd API: Operation not permitted

Wrapping the call to ioctl(UFFDIO_API) in userns_call() resolves the issue.

Fixes: #964
Signed-off-by: Mike Rapoport <rppt@linux.ibm.com>

### CODE REVIEW

File: criu/uffd.c [hunk 1]
```diff
@@ -269,10 +277,7 @@ int uffd_open(int flags, unsigned long *features)
 	if (features)
 		uffdio_api.features = *features;
 
-	if (ioctl(uffd, UFFDIO_API, &uffdio_api)) {
-		pr_perror("Failed to get uffd API");
-		goto err;
-	}
+	userns_call(uffd_api_ioctl, 0, &uffdio_api, sizeof(uffdio_api), uffd);
```
Comment (mihalicyn) [inline criu/uffd.c]: Shouldn't we check retvalue here?

Comment (rppt) [reply criu/uffd.c]: Apparently we should ;-)

**Comment by Snorch** [comment]
@avagin @xemul @rppt  Can we also merge this to master branch, or do we need separate pr? Tests are failing:
https://travis-ci.org/github/checkpoint-restore/criu/jobs/663604412

**Comment by rppt** [comment]
> @avagin @xemul @rppt Can we also merge this to master branch, or do we need separate pr? Tests are failing:
> https://travis-ci.org/github/checkpoint-restore/criu/jobs/663604412

I'm ok with either. @avagin , @xemul, it's your call

**Pull request merged**